src/main/java/com/github/dockerjava/netty/InvocationBuilder.java

```diff
@@ -262,6 +263,12 @@ public void run() {
                         channel.writeAndFlush(Unpooled.copiedBuffer(buffer, 0, read));
                     }
 
+                    // we close the writing side of the socket, but keep the read side open to transfer stdout/stderr
+                    // unfortunately, shutdownOutput is not supported by io.netty.channel.epoll.EpollDomainSocketChannel
+                    if (channel instanceof DuplexChannel) {
```

Comment:
Actually, I can change `channel` type to `DuplexChannel` because it extends `Channel` and avoid type checks. Will have to cast in `InetSocketInitializer` and `UnixDomainSocketInitializer`.